blog/16-02-2023-how-to-build-crowdfunding-platform-on-celo/how-to-build-crowdfunding-platform-on-celo.md

```diff
@@ -0,0 +1,526 @@
+
+```js
+function getTotalCauses() {
+    const causes = contract.methods.totalCauses().call();
+    return causes;
+}
+```
+This function returns all the total causes in the smart contract. It calls `totalCauses()` function of the smart contract.
+
+```js
+async function createCause(name, description, ipfsHash, target) {
+    const account = await getAccount();
+    await contract.methods.createCause(name, description, ipfsHash, target).send({ from: account });
+}
+```
+This function is called when a user presses submit button to create a cause. When this function is executed, Metamask will ask you to sign the transaction.
+
+```js
+async function fundCauseById(causeId){
+    let account = await getAccount();
+    let amount = document.getElementById(`input${causeId}`).value;
+    await contract.methods.fundCauseById(causeId).send({from: account, value: amount});
+}
+```
+This function is invoked when a user presses the `Fund` button of the cause. Whatever amount the user gives in `Wei`, CELO will be transferred towards the creator of the cause.
+
+```js
+async function fillCauseDOM() {
+    const causes = await getTotalCauses();
+    for (i = 0; i <= causes; i++) {
+        let cause = await contract.methods.causeById(i).call();
+        let causeName = cause[0];
+        let description = cause[1];
+        let ipfsHash = cause[2];
+        let targetAmount = cause[3];
+        let raisedAmount = cause[4];
+        let newDiv = document.createElement("div");
+        newDiv.setAttribute("id", i);
+        newDiv.setAttribute("class", "card-body");
+        document.getElementById('causesDOM').appendChild(newDiv);
+        newDiv.innerHTML = `<div class="card">
+                <div class="card-body">
+                    Name: ${causeName}
+                    <br />
+                    Description: ${description}
+                    <br />
+                    IFPS Hash: <a href="https://ipfs.filebase.io/ipfs/${ipfsHash}">Link</a>
+                    <br />
+                    Target Amount(in wei): ${targetAmount}
+                    <br />
+                    Raised Amount(in wei): ${raisedAmount}
+                    <br />
+                    Fund amount: <input type="text" id="input${i}">
+                    <br />
+                    <button class="btn btn-primary" onclick="fundCauseById(${i})">Fund</button>
+                </div>
+            </div>`
+        }
+}
+```
+`fillCauseDOM()` is used to iterate through all the causes and dynamically add the data to the DOM.
+
+This is how your frontend will look like if you are followed the tutorial so far:
+
+![dApp](https://i.imgur.com/fE0JX8V.png)
+
+Also, when anyone will fund a cause. This is how the transaction will look like:
+
+![Token Transfer](https://i.imgur.com/WhQLCC1.png)
+
+Also, here is the complete source code: https://github.com/avirajkhare00/celo-crowdfunding-platform
```

Comment:
Just use the correct formatting for the link it is [Github-Repo](https://github.com/avirajkhare00/celo-crowdfunding-platform)

Reply:
Done :)